crates/objc2-proc-macros/src/extern_methods.rs

```diff
@@ -2,6 +2,50 @@ use proc_macro2::TokenStream;
 use quote::ToTokens;
 use syn::parse::{Parse, ParseStream};
 
+enum RetainSemantics {
+    Init,
+    New,
+    Other,
+}
+
+// Examples:
+//
+//   #[method(sel = initTextCell:, retain = init)]
```

Comment:
Similar to my other comment, we could allow eliding the retain info here as well (by examining the type of `this`).